src/server/services/wms/qgswmsparameters.cpp

```diff
@@ -680,6 +690,41 @@ namespace QgsWms
     return mWmsParameters[ QgsWmsParameter::WIDTH ].toInt();
   }
 
+  QString QgsWmsParameters::srcHeight() const
+  {
+    return mWmsParameters[ QgsWmsParameter::SRCHEIGHT ].toString();
+  }
+
+  QString QgsWmsParameters::srcWidth() const
+  {
+    return mWmsParameters[ QgsWmsParameter::SRCWIDTH ].toString();
+  }
+
+  int QgsWmsParameters::srcHeightAsInt() const
+  {
+    return mWmsParameters[ QgsWmsParameter::SRCHEIGHT ].toInt();
+  }
+
+  int QgsWmsParameters::srcWidthAsInt() const
+  {
+    return mWmsParameters[ QgsWmsParameter::SRCWIDTH ].toInt();
+  }
+
+  int QgsWmsParameters::getHeightAsInt() const
+  {
+    if ( QStringList( { QStringLiteral( "GetLegendGraphic" ), QStringLiteral( "GetLegendGraphics" ) } ).contains( request() ) && srcHeightAsInt() > 0 )
+      return srcHeightAsInt();
+    return heightAsInt();
+  }
```

Comment:
I think that `QgsWmsParameters` should not manage requests specificities. Could you do the magic within the `QgsRenderer::getFeatureInfo` instead?